sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/queries/sql/SqlQuery5.java

```diff
@@ -57,24 +57,33 @@
 
   private static final String QUERY_TEMPLATE =
       ""
-          + " SELECT auction, num "
-          + "    FROM (SELECT B1.auction, count(*) AS num, "
-          + "       HOP_START(B1.dateTime, INTERVAL '%1$d' SECOND, "
-          + "          INTERVAL '%2$d' SECOND) AS starttime "
-          + "    FROM Bid B1 "
-          + "    GROUP BY B1.auction, "
-          + "       HOP(B1.dateTime, INTERVAL '%1$d' SECOND, "
-          + "          INTERVAL '%2$d' SECOND)) B1 "
-          + " JOIN (SELECT max(B2.num) AS maxnum, B2.starttime "
-          + "    FROM (SELECT count(*) AS num, "
-          + "       HOP_START(B2.dateTime, INTERVAL '%1$d' SECOND, "
-          + "          INTERVAL '%2$d' SECOND) AS starttime "
-          + "    FROM Bid B2 "
-          + "    GROUP BY B2.auction, "
-          + "       HOP(B2.dateTime, INTERVAL '%1$d' SECOND, "
-          + "          INTERVAL '%2$d' SECOND)) B2 "
-          + "    GROUP BY B2.starttime) B2 "
-          + " ON B1.starttime = B2.starttime AND B1.num >= B2.maxnum ";
+          + " SELECT AuctionBids.auction, AuctionBids.num "
+          + " FROM ("
+          + "   SELECT"
+          + "     B1.auction,"
+          + "     count(*) AS num,"
+          + "     HOP_START(B1.dateTime, INTERVAL '%1$d' SECOND, INTERVAL '%2$d' SECOND) AS starttime"
+          + "   FROM Bid B1 "
+          + "   GROUP BY "
+          + "     B1.auction,"
+          + "     HOP(B1.dateTime, INTERVAL '%1$d' SECOND, INTERVAL '%2$d' SECOND)"
+          + " ) AS AuctionBids"
+          + " JOIN ("
+          + "   SELECT "
+          + "     max(CountBids.num) AS maxnum, "
+          + "     HOP_START(CountBids.starttime, INTERVAL '%1$d' SECOND, INTERVAL '%2$d' SECOND) AS starttime"
+          + "   FROM ("
+          + "     SELECT"
+          + "       count(*) AS num,"
+          + "       HOP_START(B2.dateTime, INTERVAL '%1$d' SECOND, INTERVAL '%2$d' SECOND) AS starttime"
+          + "     FROM Bid B2 "
+          + "     GROUP BY "
+          + "       B2.auction, "
+          + "       HOP(B2.dateTime, INTERVAL '%1$d' SECOND, INTERVAL '%2$d' SECOND)"
+          + "     ) AS CountBids"
+          + "   GROUP BY HOP(CountBids.starttime, INTERVAL '%1$d' SECOND, INTERVAL '%2$d' SECOND)"
```

Comment:
This additional introduction of a windowed GROUP BY causes the output to differ.